pkg/storage/buckets/store.go

```diff
@@ -166,27 +182,82 @@ func (b *BucketStore) List(ctx context.Context, startAfter, endBefore string, li
 		pet, err := strconv.Atoi(pathEncType)
 		if err != nil {
-			return Meta{}, err
+			return Meta{}, errs.New("invalid metadata field for path-enc-type: %v", err)
+		}
+		out.PathEncryptionType = storj.Cipher(pet)
+	}
+	if defaultSegSize := m.UserDefined["default-seg-size"]; defaultSegSize != "" {
+		dss, err := strconv.ParseInt(defaultSegSize, 10, 64)
+		if err != nil {
+			return Meta{}, errs.New("invalid metadata field for path-enc-type: %v", err)
+		}
+		out.SegmentsSize = dss
+	}
+	if defaultEncType := m.UserDefined["default-enc-type"]; defaultEncType != "" {
+		det, err := strconv.Atoi(defaultEncType)
+		if err != nil {
+			return Meta{}, errs.New("invalid metadata field for default-enc-type: %v", err)
+		}
+		out.EncryptionScheme.Cipher = storj.Cipher(det)
+	}
+	if defaultEncBlockSize := m.UserDefined["default-enc-blksz"]; defaultEncBlockSize != "" {
+		deb, err := strconv.ParseInt(defaultEncBlockSize, 10, 32)
+		if err != nil {
+			return Meta{}, errs.New("invalid metadata field for default-enc-blksz: %v", err)
+		}
+		out.EncryptionScheme.BlockSize = int32(deb)
+	}
+	if defaultRSAlgo := m.UserDefined["default-rs-algo"]; defaultRSAlgo != "" {
+		dra, err := strconv.Atoi(defaultRSAlgo)
+		if err != nil {
+			return Meta{}, errs.New("invalid metadata field for default-rs-algo: %v", err)
+		}
+		out.RedundancyScheme.Algorithm = storj.RedundancyAlgorithm(dra)
+	}
+	if defaultRSShareSize := m.UserDefined["default-rs-sharsz"]; defaultRSShareSize != "" {
+		drs, err := strconv.ParseInt(defaultRSShareSize, 10, 32)
+		if err != nil {
+			return Meta{}, errs.New("invalid metadata field for default-rs-sharsz: %v", err)
+		}
+		out.RedundancyScheme.ShareSize = int32(drs)
+	}
+	if defaultRSRequired := m.UserDefined["default-rs-reqd"]; defaultRSRequired != "" {
+		drr, err := strconv.ParseInt(defaultRSRequired, 10, 16)
+		if err != nil {
+			return Meta{}, errs.New("invalid metadata field for default-rs-reqd: %v", err)
+		}
+		out.RedundancyScheme.RepairShares = int16(drr)
+	}
+	if defaultRSRepairThresh := m.UserDefined["default-rs-repair"]; defaultRSRepairThresh != "" {
+		drr, err := strconv.ParseInt(defaultRSRepairThresh, 10, 16)
+		if err != nil {
+			return Meta{}, errs.New("invalid metadata field for default-rs-repair: %v", err)
+		}
+		out.RedundancyScheme.RepairShares = int16(drr)
+	}
+	if defaultRSOptimal := m.UserDefined["default-rs-optim"]; defaultRSOptimal != "" {
+		drr, err := strconv.ParseInt(defaultRSOptimal, 10, 16)
+		if err != nil {
+			return Meta{}, errs.New("invalid metadata field for default-rs-optim: %v", err)
+		}
+		out.RedundancyScheme.RepairShares = int16(drr)
+	}
+	if defaultRSTotal := m.UserDefined["default-rs-total"]; defaultRSTotal != "" {
+		drr, err := strconv.ParseInt(defaultRSTotal, 10, 16)
+		if err != nil {
+			return Meta{}, errs.New("invalid metadata field for default-rs-total: %v", err)
 		}
-		cipher = storj.Cipher(pet)
+		out.RedundancyScheme.RepairShares = int16(drr)
```

Comment:
This value should be assigned to `TotalShares`, not to `RepairShares`.

Reply:
wow, i must have been half asleep when i did all this